base/parse.jl

```diff
@@ -240,6 +240,32 @@ float(x::AbstractString) = parse(Float64,x)
 
 float(a::AbstractArray{<:AbstractString}) = map!(float, similar(a,typeof(float(0))), a)
 
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
+    s = strip(s)
+    sign = s[1] == '-' ? -1 : 1
+    sign == -1 && (s = s[2:end])
```

Comment:
need to handle a leading `+` too

Reply:
Also, this is not type-stable, since slicing may not return the same type of string.  And this allocates a copy of the string — I think that `parse` should always be non-allocating.